### DISCUSSION TITLE
HomeKit connection status

### DISCUSSION THREAD

**Discussion by CK4444** [HomeSpan/HomeSpan#332] [Ideas]
Hi, 
I have a Temperature Sensor running using a DHT22 connected to a WIMOS D1 Mini ESP 32. After running a couple of hours the connection gets lost (no data in HomeKit) and the assumption was, as experiences show, the DHT22 having a problem. I included a few lines to send the temperature data to my IOT Cloud  service and - surprise - the DHT22 was working properly, sensor values coming regular without drops. 
In order to deeper analyse the reasons for the connection losses my question: Is there any kind of status information on the connection to HomeKit available via the Homespan API ? 
(Btw, when these connections are lost, the WEB Server with the log does not respond as well). 
Thanks a lot for any input, 
Christian

**Comment by HomeSpan** [comment]
Unfortunately there's not a lot of built-in options for debugging since the Home App controls all connectivity.  However, it's puzzling that you are also seeing the Web Log also freeze - is it the case that there are no new log messages, or is it completely non-responsive?  Can you type commands into the the CLI, such as 'i' or 's', or is the CLI non-responsive as well?  If the CLI is working (which means HomeSpan is operating correctly but without connectivity) perhaps what are you using to send IOT Cloud message is somehow interfering?

**Reply by CK4444** [reply]
- The web Log is not responsive at all
- This happens, when I run the controller stand alone (USB for power only) - therefore I cannot say something about the CLI
- I use the http library to send a simple GET request to thingspeak. The problem occurred already before I implemented this - that's why I did it ( to check, whether the controller is working at all)

**Comment by HomeSpan** [comment]
The other thing to try is turn off the clock-sync for the weblog.  You can do this by setting the *timeServerURL* to NULL (the second parameter) in the `homeSpan.enableWebLog()` method.

**Reply by CK4444** [reply]
Thanks,
I will try.
